obspy/core/util/headers.py

```diff
@@ -0,0 +1,484 @@
+# -*- coding: utf-8 -*-
+"""
+Base constants for ObsPy.
+
+:copyright:
+    The ObsPy Development Team (devs@obspy.org)
+:license:
+    GNU Lesser General Public License, Version 3
+    (https://www.gnu.org/copyleft/lesser.html)
+"""
+# NO IMPORTS FROM OBSPY OR FUTURE IN THIS FILE! (file gets used at
+# installation time)
+import sys
+
+
+# XXX TODO when bumping to numpy 1.9.0: replace bytes() in io.reftek with
+# np.tobytes()
+HARD_DEPENDENCIES = {
+    "future": ">=0.12.4",
+    "numpy": ">=1.6.1",
+    "scipy": ">=0.9.0",
+    "matplotlib": ">=1.1.0",
+    "lxml": "",
+    "setuptools": "",
+    "sqlalchemy": "",
+    "decorator": "",
+    "requests": ""
+    }
+OPTIONAL_DEPENDENCIES = {
+    'arclink': {
+        "cryptography": "",
+        "m2crypto": "",
+        "pycrypto": ""},
+    # for docs, also see conda environment file under
+    # misc/docs/py3-docs-master.yml
+    # other versions might work as well but these are currently used in our
+    # docs buildbot and therefore confirmed to work
+    'docs': {
+        "sphinx": "==1.2.2",
```

Comment:
I'm not sure that people actually use those optional dependency specifiers with pip. I'm mostly thinking of this as a place to summarize all hard/soft dep versions and to keep track of things. With building docs it can be really tough to get the correct versions installed, so I'd always happily try to use versions that are confirmed to work.

On the other hand, there's the conda environment file with exact versions that our docs buildbot uses, so that's enough I guess and people could also look it up there.